packages/wallet/src/cip30.ts

```diff
@@ -406,7 +406,10 @@ const baseCip30WalletApi = (
       scope.dispose();
     }
   },
-  signData: async (addr: Cardano.PaymentAddress | Bytes, payload: Bytes): Promise<Cip30DataSignature> => {
+  signData: async (
+    addr: Cardano.PaymentAddress | Cardano.DRepID | Bytes,
+    payload: Bytes
+  ): Promise<Cip30DataSignature> => {
     logger.debug('signData');
     const hexBlobPayload = HexBlob(payload);
     const signWith = Cardano.PaymentAddress(addr);
```

Comment:
I think it would be clearer if we use the DrepID type. It would also require updating `Cip30SignDataRequest` to accept `DrepID`.
```diff
- const signWith = Cardano.PaymentAddress(addr);
+ const signWith = DRepID.isValid(addr) ? DRepID(addr) : Cardano.PaymentAddress(addr);

Reply:
Should I also add a `DRepID.canSign()` function to check if the DRepID is the correct type?

According to CIP-95 only type

Reply:
Otherwise, it lets me sign with types 6, 7, 14 and 15. The rest fail because of `Invalid address` in `getAddressBytes() (cip30signData.ts)`

Reply:
Added this and addressed the rest of the comments in https://github.com/input-output-hk/cardano-js-sdk/pull/907/commits/c74bed59c4b80616b1cc71167cb67ac7656f2921